### PR TITLE
Docker container for serving static content

### LINKED ISSUE #614: Add a Docker container type for serving static content -  React, etc.
We plan on using Nginx to serve the content

### PR DESCRIPTION
**Pull request by pshomov**
Fixes #614 

Added a dummy react application to be able to test the Docker image building. Will remove it before merging to master.

### CODE REVIEW

File: scripts/_docker.sh [hunk 1]
```diff
@@ -9,12 +9,19 @@ APP_HOME=`cat $PROJECT_ROOT/workspace.json | jq ".projects[\"$APP\"].root" -r`
...
-else
-    # Just build the container but do not publish it to the registry
-    PUBLISH_TO_REGISTRY=""
-fi
+case $PUBLISH in
+    true)
+        PUBLISH_TO_REGISTRY="--push"
+        ;;
+    local)
+        PUBLISH_TO_REGISTRY="--load"
```
Comment (sindrig) [inline scripts/_docker.sh:17]: nice

File: scripts/docker-static.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -euxo pipefail
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+source $DIR/_common.sh
+
+# Build Docker images for Next.js-based apps
```
Comment (sindrig) [inline scripts/docker-static.sh:8]: not next.js right?

File: scripts/Dockerfile [hunk 3]
```diff
@@ -88,3 +88,20 @@ COPY --from=builder --chown=runner:runners /build/${APP_HOME}/public /webapp/pub
...
+
+FROM nginx:1.19.2-alpine as output-static
+
+ARG GIT_BRANCH
+ARG GIT_SHA
+LABEL branch=${GIT_BRANCH}
+LABEL commit=${GIT_SHA}
+ENV GIT_BRANCH=${GIT_BRANCH}
+ENV GIT_SHA=${GIT_SHA}
+ARG APP
```
Comment (sindrig) [inline scripts/Dockerfile:100]: minor but i like all `ARG`s at the top so you immediately see what build args you need

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -341,6 +351,45 @@ jobs:
         if: ${{ matrix.projects != 'dummy' }}
         run: ./scripts/run-in-parallel.sh docker-express
 
+  docker-static:
```
Comment (sindrig) [inline .github/workflows/build.yml:354]: we're up to 3 almost identical blocks (docker-static, docker-express and docker-next) might want to start thinking about extracting and re-using bits and pieces.

**Review by sindrig — APPROVED** [review]

File: .github/workflows/build.yml (hunk 4 above)
Comment (pshomov) [reply .github/workflows/build.yml:354]: Agreed and made the same comment in 3d6ac62b08c8db1e18a6a974ef13ad959ae36246. But the CI needs a bit of cleaning up anyway after the iterations it went through.

File: scripts/docker-static.sh (hunk 2 above)
Comment (pshomov) [reply scripts/docker-static.sh:8]: fixed

File: scripts/Dockerfile (hunk 3 above)
Comment (pshomov) [reply scripts/Dockerfile:100]: fixed

**Review by sindrig — APPROVED** [review]
LGTM :partying_face:

**Pull request merged**